### ISSUE TITLE
We can extract the schema_name and schema_version from the passed schemas in the `verify_presentation` method and pass them down (`verify_presentation`  -> `verify_requested_restrictions` -> `gather_filter_info`).

### ISSUE THREAD

**Issue by blu3beri** [hyperledger/anoncreds-rs#34]
We can extract the schema_name and schema_version from the passed schemas in the `verify_presentation` method and pass them down (`verify_presentation`  -> `verify_requested_restrictions` -> `gather_filter_info`).

For the other ones, maybe we need to look at adding an `issuerId` property to anoncreds objects after all? I think there's no harm in it, and it would be backwards compatible with current anoncreds (issuerId is always the indy did in that case). Relevant issue: https://github.com/hyperledger/anoncreds-spec/issues/102

For the indy method it would be the indy did, for cheqd it would be the cheqd did. For non ledger based methods there must be some point of reference of who created the schema/credential definition. For https it could be the host (e.g. schema `https://hyperledger.org/membershipSchema` would have `issuerId` of `https://hyperledger.org`

- [ ] support both `_did` and `id` for restrictions

@swcurran thoughts?

_Originally posted by @TimoGlastra in https://github.com/hyperledger/anoncreds-rs/pull/25#discussion_r1043001139_

**Comment by swcurran** [comment]
An interesting question.  Agree that this is relevant to Issue 102 in the Spec repo.

I think we should include it.  It would also be good if we could validate it, but that gets tricky -- AnonCreds Method specific and even then, self-asserted. Would it be possible to require that all AnonCreds URIs stem from the `issuerId`?  That would work for the existing/planned methods but would be hard to explain and enforce.

Bottom line -- I think we should include it and require it.

**Comment by blu3beri** [comment]
Okay I spoke a bit too quickly and just saw @TimoGlastra PR https://github.com/hyperledger/anoncreds-spec/pull/126. Will use that as a reference.
